--- libosip2-4.0.0/src/osip2/Makefile.am~	2010-01-06 14:58:16.000000000 +0000
+++ libosip2-4.0.0/src/osip2/Makefile.am	2012-12-17 06:01:21.217792703 +0000
@@ -13,11 +13,11 @@
 libosip2_la_SOURCES+=port_sema.c port_thread.c port_condv.c
 endif
 
-libosip2_la_LDFLAGS = -version-info $(LIBOSIP_SO_VERSION) \
- $(FSM_LIB) $(EXTRA_LIB) ../osipparser2/libosipparser2.la -no-undefined
+libosip2_la_LDFLAGS = -version-info $(LIBOSIP_SO_VERSION) -no-undefined
+libosip2_la_LIBADD = $(FSM_LIB) $(EXTRA_LIB) ../osipparser2/libosipparser2.la 
 
 
 INCLUDES = -I$(top_srcdir)/include
-AM_CFLAGS = $(SIP_CFLAGS) $(SIP_FSM_FLAGS) $(SIP_EXTRA_FLAGS)
+AM_CFLAGS = $(SIP_CFLAGS) $(SIP_FSM_FLAGS) $(SIP_EXTRA_FLAGS) $(PTHREAD_CFLAGS)
 
 
--- libosip2-4.0.0/src/osip2/Makefile.in~	2012-03-05 20:15:06.000000000 +0000
+++ libosip2-4.0.0/src/osip2/Makefile.in	2012-12-17 06:04:38.063790228 +0000
@@ -70,7 +70,7 @@
   sed '$$!N;$$!N;$$!N;$$!N;s/\n/ /g'
 am__installdirs = "$(DESTDIR)$(libdir)"
 LTLIBRARIES = $(lib_LTLIBRARIES)
-libosip2_la_LIBADD =
+libosip2_la_LIBADD = $(FSM_LIB) $(EXTRA_LIB) ../osipparser2/libosipparser2.la
 am__libosip2_la_SOURCES_DIST = ict_fsm.c ist_fsm.c nict_fsm.c \
 	nist_fsm.c ict.c ist.c nict.c nist.c fsm_misc.c osip.c \
 	osip_transaction.c osip_event.c port_fifo.c osip_dialog.c \
@@ -235,11 +235,10 @@
 	ist.c nict.c nist.c fsm_misc.c osip.c osip_transaction.c \
 	osip_event.c port_fifo.c osip_dialog.c osip_time.c \
 	$(am__append_1)
-libosip2_la_LDFLAGS = -version-info $(LIBOSIP_SO_VERSION) \
- $(FSM_LIB) $(EXTRA_LIB) ../osipparser2/libosipparser2.la -no-undefined
+libosip2_la_LDFLAGS = -version-info $(LIBOSIP_SO_VERSION) -no-undefined
 
 INCLUDES = -I$(top_srcdir)/include
-AM_CFLAGS = $(SIP_CFLAGS) $(SIP_FSM_FLAGS) $(SIP_EXTRA_FLAGS)
+AM_CFLAGS = $(SIP_CFLAGS) $(SIP_FSM_FLAGS) $(SIP_EXTRA_FLAGS) $(PTHREAD_CFLAGS)
 all: all-am
 
 .SUFFIXES:
